torch/testing/_internal/common_utils.py

```diff
@@ -398,6 +398,18 @@ def wrapper(*args, **kwargs):
             fn(*args, **kwargs)
     return wrapper
 
+# RAII guard for setting deterministic flag and automatically resetting it
+# to its original value
+class DeterministicGuard:
```

Comment:
I was expecting this to be modeled as a context manager with dunder enter and dunder exit. What would you think of that to be more Pythonic?

Reply:
Makes sense, I've updated it